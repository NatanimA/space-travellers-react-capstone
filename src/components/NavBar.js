import { NavLink } from 'react-router-dom';
import logo from '../brand.png';

const NavBar = () => (
  <header>
    <div>
      <img className="brand-logo" src={logo} alt="Brand" />
      <h1>Space Traveler&apos;s Hub</h1>
    </div>
    <nav>
      <ul>
        <li>
          <NavLink to="/"> Rockets</NavLink>
        </li>

        <li className="nav-cutter">
          <NavLink to="/missions">Missions</NavLink>
        </li>

        <li>
          <NavLink className="my-profile" to="/profile">My profile</NavLink>
        </li>

      </ul>
    </nav>
  </header>
);

export default NavBar;
