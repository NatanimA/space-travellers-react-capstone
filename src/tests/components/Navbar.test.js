import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import NavBar from '../../components/NavBar';

describe('Tests NavBar Component', () => {
  it('Should render NavBar Component', () => {
    const tree = renderer.create(<BrowserRouter><NavBar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(' Checks Links and Active links', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    const list = screen.getByRole('list').childElementCount;
    expect(list).toEqual(3);
    const link = screen.getByText('Missions');
    userEvent.click(link);
    expect(link.closest('a')).toHaveAttribute('class', 'active');
  });
});
