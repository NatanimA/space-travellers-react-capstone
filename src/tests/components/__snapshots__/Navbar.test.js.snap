// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests NavBar Component Should render NavBar Component 1`] = `
<header>
  <div>
    <img
      alt="Brand"
      className="brand-logo"
      src="brand.png"
    />
    <h1>
      Space Traveler's Hub
    </h1>
  </div>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
           Rockets
        </a>
      </li>
      <li
        className="nav-cutter"
      >
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="my-profile"
          href="/profile"
          onClick={[Function]}
        >
          My profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
